src/modeler/AutoRest.Swagger/Validation/ServiceDefinitionParameters.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using AutoRest.Core.Logging;
+using AutoRest.Core.Properties;
+using AutoRest.Core.Validation;
+using AutoRest.Swagger.Model;
+
+namespace AutoRest.Swagger.Validation
+{
+    public class ServiceDefinitionParameters : TypedRule<Dictionary<string, SwaggerParameter>>
+    {
+        private const string SubscriptionId = "subscriptionId";
+        private const string ApiVersion = "api-version";
+        /// <summary>
+        /// This rule passes if the parameters section contains both subscriptionId and api-version parameters
+        /// </summary>
+        /// <param name="paths"></param>
+        /// <returns></returns>
+        public override bool IsValid(Dictionary<string, SwaggerParameter> ParametersMap)
+        {
+            if (ParametersMap.Count==0 || ParametersMap == null)
```

Comment:
Actually, you don't want to check the `count`; that presumes either that the dictionary implementation knows the actual item count (which not all do),  or you're forcing the dictionary to count the items. The iteration below will be nearly 0 time if the collection is empty anyway.

So replace this with 

``` csharp
if( ParametersMap == null ) 
{ 
    return false;
}
```